libtiledbsoma/src/utils/arrow_adapter.cc

```diff
@@ -108,17 +154,20 @@ std::unique_ptr<ArrowSchema> ArrowAdapter::arrow_schema_from_tiledb_array(
     auto nattr = tiledb_schema.attribute_num();
 
     std::unique_ptr<ArrowSchema> arrow_schema = std::make_unique<ArrowSchema>();
-    arrow_schema->format = "+s";
+    arrow_schema->format = strdup("+s");
```

Comment:
I know I'm the one who added this but I'm not sure that it's necessary here. I may have overcorrected.

Reply:
I think it is. It is just stylistic. If you peek into what `nanoarrow` does it actually allocates explicitly (as do I over in tiledb-r)

```c
ArrowErrorCode ArrowSchemaSetFormat(struct ArrowSchema* schema, const char* format) {
  if (schema->format != NULL) {
    ArrowFree((void*)schema->format);
  }

  if (format != NULL) {
    size_t format_size = strlen(format) + 1;
    schema->format = (const char*)ArrowMalloc(format_size);
    if (schema->format == NULL) {
      return ENOMEM;
    }

    memcpy((void*)schema->format, format, format_size);
  } else {
    schema->format = NULL;
  }

  return NANOARROW_OK;
}
```

This is different from what it also has 'view' inits for schema which just borrow the pointed to memory because the contract the viewer can assume the viewed object exists.

In short, your `strdup()` was fine.  And after 30+ plus years of C it possibly taught me another K+R (at style) function so all good!